### ISSUE TITLE
Flux native select value can't be set to `0`

### ISSUE THREAD

**Issue by gustavosobrinho01** [livewire/flux#907]
### Flux version

v1.0.29

### Livewire version

v3.5.16

### What is the problem?

When using the select as shown in the code and images below, the screen loads with the value 0. When I select the “Ativos” option, the attribute value changes to 1. However, when switching back to the “All Status” option, the value should reset to 0, but as shown in the image, it displays as “Todos+os+Status”.

<img width="183" alt="Image" src="https://github.com/user-attachments/assets/8fe2d4b5-5473-4bb8-8a6d-64c3b42d494a" />
<img width="204" alt="Image" src="https://github.com/user-attachments/assets/0b353aea-4a7d-43fa-a466-8dc372f2dd10" />

### Code snippets

```
<div>
    <flux:select wire:model="{{ $attributes->wire('model')->value }}"
                 wire:change="{{ $attributes->wire('change')->value }}">
        <flux:option value="0">Todos os Status</flux:option>
        <flux:option value="1">Ativos</flux:option>
        <flux:option value="2">Excluídos</flux:option>
    </flux:select>
</div>
```

```
<x-filters.status wire:model="statusSearch" wire:change="filter" autofocus/>
```

### How do you expect it to work?

I expect the value in the URI to be set to 0, not “Todos+os+Status”.

### Please confirm (incomplete submissions will not be addressed)

- [x] I have provided easy and step-by-step instructions to reproduce the bug.
- [x] I have provided code samples as text and NOT images.
- [x] I understand my bug report will be closed if I haven't met the criteria above.

**Comment by joshhanley** [comment]
@gustavosobrinho01 thanks for reporting! Can you try using a value other than `0` and see if that works? I suspect there as an issue with how the select is handling a zero value

**Comment by gustavosobrinho01** [comment]
With another value, it works, but it should also work with 0, as another value does not meet my needs.

**Comment by joshhanley** [comment]
@gustavosobrinho01 yeah agreed, I just wanted to make sure it wasn’t an issue with the first item in the select vs a problem with a 0 value

**Comment by thobiasvicente** [comment]
I'm experiencing the same issue. 

This behavior is impacting the filter functionality since the expected numeric value isn't being properly set in the URL parameters.

Environment where I reproduced:
- Flux: v1.1.2
- Livewire: v3.5.18

**Comment by jeffchown** [comment]
@gustavosobrinho01 If you change your `components/filters/status.blade.php` to:

```
<div>
    <flux:select {{ $attributes }}>
        <flux:option value="0">Todos os Status</flux:option>
        <flux:option value="1">Ativos</flux:option>
        <flux:option value="2">Excluídos</flux:option>
    </flux:select>
</div>
```

and your view code to:

```
<x-filters.status wire:model.number="statusSearch" wire:change.number="filter" autofocus/>
```

does it make any difference?

**Comment by gustavosobrinho01** [comment]
@jeffchown acontece o mesmo problema.

**Comment by joshhanley** [comment]
Thanks @gustavosobrinho01 and @jeffchown! I've submitted a PR with a fix.

**Comment by jeffchown** [comment]
Thanks, @joshhanley

**Comment by gustavosobrinho01** [comment]
@joshhanley Thanks, but I didn't find this PR and I ran the update and nothing came.

**Comment by jeffchown** [comment]
@gustavosobrinho01 Josh's PR won't be in the Flux public repo. And, it won't be available via `composer update` until Caleb merges the PR into a new Flux Pro release.

**Comment by shaneniebergall** [comment]
Same issue here - would love to get this merged into the public repo asap.

**Comment by joshhanley** [comment]
@gustavosobrinho01 and @shaneniebergall thanks for the support! But please be patient, I only submitted the fix yesterday 😅 You need to give Caleb time to review and for a new release to be tagged, he is currently off sick though, so it may be a little while. This issue will be closed if Caleb merges the PR, so you should know once it has been actioned. Then it will be in the release following that, whenever that happens to be 🙂
